junit_tests/testTimeDelta.java

```diff
@@ -0,0 +1,423 @@
+        return createDelta(empty_kwargs, values);
+    }
+
+    private TimeDelta createDelta(Map<java.lang.String, org.python.Object> kwargs, int... values) {
+        org.python.Object[] args = new org.python.Object[values.length];
+
+        for (int i = 0; i < values.length; i++) {
+            args[i] = getInt(values[i]);
+        }
+
+        return new TimeDelta(args, kwargs);
+    }
+
+    private TimeDelta createDelta(float... values) {
+        org.python.Object[] args = new org.python.Object[values.length];
+
+        for (int i = 0; i < values.length; i++) {
+            args[i] = new org.python.types.Float(values[i]);
+        }
+
+        return new TimeDelta(args, empty_kwargs);
+    }
+
+    @Test
+    public void testConstructorOneArguments() {
+        TimeDelta delta = createDelta(1);
+
+        assertEquals(getInt(1), delta.days);
+        assertEquals(getInt(0), delta.seconds);
+        assertEquals(getInt(0), delta.microseconds);
+    }
+
+    @Test
+    public void testConstructorTwoArguments() {
+        TimeDelta delta = createDelta(1, 2);
+
+        assertEquals(getInt(1), delta.days);
+        assertEquals(getInt(2), delta.seconds);
+        assertEquals(getInt(0), delta.microseconds);
+    }
+
+    @Test
+    public void testConstructorThreeArguments() {
+        TimeDelta delta = createDelta(1, 2, 3);
+
+        assertEquals(getInt(1), delta.days);
+        assertEquals(getInt(2), delta.seconds);
+        assertEquals(getInt(3), delta.microseconds);
+    }
+
+    @Test
+    public void testConstructorBadArguments() {
+        org.python.Object[] args = new org.python.Object[]{getBool(false)};
+        Exception exception = assertThrows(org.python.exceptions.TypeError.class, () -> {
+            TimeDelta delta = new TimeDelta(args, empty_kwargs);
+        });
+
+        assertEquals("unsupported type for timedelta days component: bool", exception.getMessage());
+    }
+
+    @Test
+    public void testConstructorTooManyArguments() {
+        Exception exception = assertThrows(org.python.exceptions.TypeError.class, () -> {
+            TimeDelta delta = createDelta(5, 5, 5, 5, 5, 5, 5, 5);
+        });
+
+        assertEquals("__new__() takes at most 7 arguments (8 given)", exception.getMessage());
+    }
+
+    @Test
+    public void testConstructorKwargs() {
+        Map<java.lang.String, org.python.Object> kwargs = new HashMap<>();
+        kwargs.put("days", getInt(1));
+        kwargs.put("seconds", getInt(2));
+        kwargs.put("microseconds", getInt(3));
+        kwargs.put("weeks", getInt(4));
+        kwargs.put("hours", getInt(5));
+        kwargs.put("minutes", getInt(6));
+        kwargs.put("milliseconds", getInt(7));
+        TimeDelta delta = createDelta(kwargs);
+
+        assertEquals(getInt(1 + 4 * 7), delta.days);
+        assertEquals(getInt(2 + 5 * 3600 + 6 * 60), delta.seconds);
+        assertEquals(getInt(3 + 7 * 1000), delta.microseconds);
+    }
+
```

Comment:
A valid mixed args/kwargs test would be nice.
E.g. `timedelta(1, 2, weeks=5)`

Reply:
Added in 6296ecf401e592553c038b6f16ab6e1e2299af73! 🔥